presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -1010,6 +1010,16 @@ public void testDecimalRangeFilters()
         }
     }
 
+    @Test
+    public void testLambda()
+    {
+        assertQuery("select transform(x, i->i*y) from (select x, y*y as y from (values row(array[1], 2)) t(x, y))");
+
+        // test nested lambda
+        assertQuery("select transform(transform(x, i->i*z), i->i*y) from (select x, y*y as y, z*z as z from (values row(array[1], 2, 3)) t(x, y, z))");
+        assertQuery("select transform(x, i->transform(i, j->j*y)) from (select x, y*y as y from (values row(array[array[1]], 2)) t(x, y))");
+    }
+
```

Comment:
@pranjalssh 
Would this solely run on the coordinator, since everything is constant?
Would this hit the worker?
Could we use queries that actually use some tables?

Reply:
VALUES will make it run on the worker.